src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -3348,6 +3363,15 @@ namespace SurfaceGeometry {
         int numSides;
         Real64 SchedMinValue;
         Real64 SchedMaxValue;
+
+        if (state.dataSolarShading->GetInputFlag) {
+            SolarShading::GetShadowingInput(state);
+            state.dataSolarShading->GetInputFlag = false;
+            state.dataSolarShading->MaxHCV =
+                (((max(15, state.dataSurface->MaxVerticesPerSurface) + 16) / 16) * 16) - 1; // Assure MaxHCV+1 is multiple of 16 for 128 B alignment
+            assert((state.dataSolarShading->MaxHCV + 1) % 16 == 0);
+        }
+
         auto &cCurrentModuleObject = state.dataIPShortCut->cCurrentModuleObject;
```

Comment:
This is needed since in the original running sequence, the shadow calculation module inputs are not processed yet when the surface geometry input processing is conducted. Therefore, need to first process the shadow calculation inputs first. Flag is set so this operation does not need to duplicated in later encounters.

Reply:
After the last batch of changes and comments: Except now that it is calling a stripped-down version of input processing other than the full-blown one.

Reply:
Rather than adding a new function to partially process the ShadowCalculation method, it would be better to move the check to somewhere else after the normal input processing occurs. A simple approach would be to set a single flag values for something like `state.dataSolarShading->anyScheduledShadingSurface` and check that at the time PixelCounting is processed in it's full getinput.  I don't think it's necessary to warn for every shading surface that has a schedule.